trl/trainer/grpo_trainer.py

```diff
@@ -751,6 +752,15 @@ def _generate_and_score_completions(
         eos_idx[is_eos.any(dim=1)] = is_eos.int().argmax(dim=1)[is_eos.any(dim=1)]
         sequence_indices = torch.arange(is_eos.size(1), device=device).expand(is_eos.size(0), -1)
         completion_mask = (sequence_indices <= eos_idx.unsqueeze(1)).int()
+        # Identify truncated samples (completions that reached max_completion_length without generating an EOS token)
+        truncated_samples = torch.zeros(is_eos.size(0), dtype=torch.bool, device=device)
+        if self.mask_truncated_samples:
+            # A sample is truncated if it has no EOS token and its length equals max_completion_length
+            truncated_samples = ~is_eos.any(dim=1) & (completion_ids.size(1) == self.max_completion_length)
+            # Log the % of truncated samples 
+            truncated_pct = truncated_samples.float().mean().item() * 100
```

Comment:
If I'm not mistaken, we already have this metric as `completions/clipped_ratio` from #3188